### PR TITLE
Handle dm-verity-protected filesystems in s390x secex path

### PR DESCRIPTION
**Pull request by jlebon**
Instead of shipping the rootfs and bootfs using LUKS, we're pivoting to
verifying integrity using dm-verity instead.

Add `veritysetup` to the OS. Teach `transposefs` to transfer the
filesystem over dm-verity. Delete the root and boot hash partitions.
Delete some of the bits we previously added for LUKS enablement.

For more information, see:
https://github.com/coreos/coreos-assembler/pull/3039

### CODE REVIEW

**Comment by jlebon** [comment]
Requires: https://github.com/coreos/coreos-assembler/pull/3039

**Review by cgwalters — DISMISSED** [review]

**Review by nikita-dubrovskii — DISMISSED** [review]

**Comment by nikita-dubrovskii** [comment]
Tested on FCOS and RHCOS - everything works

**Comment by jlebon** [comment]
Added a commit message to the last commit and marked ready for review now that the cosa PR merged!

**Review by nikita-dubrovskii — APPROVED** [review]

**Review by nikita-dubrovskii — APPROVED** [review]

File: overlay.d/05core/usr/lib/dracut/modules.d/40ignition-ostree/ignition-ostree-transposefs.sh [hunk 1]
```diff
@@ -88,9 +100,23 @@ mount_and_save_filesystem_by_label() {
     local label=$1; shift
     local saved_fs=$1; shift
     local fs=/dev/disk/by-label/${label}
+    if [[ -f /run/coreos/secure-execution ]]; then
+        local roothash_karg=${label}fs.roothash
+        local roothash=$(karg "${roothash_karg}")
+        if [ -z "${roothash}" ]; then
+          echo "Missing kernel argument ${roothash_karg}; aborting"
```
Comment (bgilbert) [inline overlay.d/05core/usr/lib/dracut/modules.d/40ignition-ostree/ignition-ostree-transposefs.sh:107]: Nit: indentation.

**Review by bgilbert — APPROVED** [review]
LGTM!

**Pull request merged**